docs/types.rst

```diff
@@ -0,0 +1,23 @@
+.. currentmodule:: jsonschema
+
+=============
+Type Checking
+=============
+
+Each :class:`IValidator` has an associated :class:`TypeChecker`. The
```

Comment:
So, half baked thought, but I guess in retrospect it's not even obvious that every validator *has* the `type` validator. In terms of JSON Schema it seems pretty likely that it will stay forever, but in the more generic notion, a bit less so... Maybe someone creates a validator that doesn't have that property (http://github.com/Julian/Seep is already such an example I think), in which case providing a type_checker to create is a bit random...

Definitely not saying to change anything, but feel the need to note that uncomfortableness somewhere.

Reply:
Interesting use case. I agree that it seems pretty core to JSON schema. Nonetheless I've reworded the docs slightly. It is slightly clunky but one would simply not provide a `TypeChecker` in such a case.